ruby-challenges.rb

```diff
@@ -3,33 +3,69 @@
 
 # --------------------1) Create a method that takes in an array of words and a single letter and returns all the words that contain that particular letter.
 
-beverages_array = ['coffee', 'tea', 'juice', 'water', 'soda water']
 
 letter_o = 'o'
 # Expected output: ['coffee', 'soda water']
 letter_t = 't'
 # Expected output: ['tea', 'water', 'soda water']
 
 
-# -------------------2) Create a method that takes in an array of numbers and returns the sum of the numbers. Use the test variables provided.
 
+
+def find_words_with_letter(array, string_letter)
+  array.select { |word| word.include?(string_letter) }
+end
+
+p find_words_with_letter(beverages_array, letter_o)
+p find_words_with_letter(beverages_array, letter_t)
+
+
+
+# -------------------2) Create a method that takes in an array of numbers and returns the sum of the numbers. Use the test variables provided.
+#
 nums_array1 = [42, 7, 27]
 # Expected output: 76
 
+p nums_array1.inject(0){|sum,x| sum + x }
+
+#
 nums_array2 = [25, 17, 47, 11]
 # Expected output: 100
 
+p nums_array2.inject(0){|sum,x| sum + x }
```

Comment:
Inject is a fun method. Remember that the goal is to great reusable code - a custom method where you could pass any given array.